### ISSUE TITLE
Ensure a reconciliation is triggered if the license changed

### ISSUE THREAD

**Issue by barkbay** [elastic/cloud-on-k8s#4152]
Some of our controllers should be run when the license is changed. For example it should be the case for the remote cluster controller.

It used to be the purpose of the [NewWatch](https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/common/license/watches.go#L14) function:

```golang
func NewWatch(fn handler.Mapper) watches.HandlerRegistration {
	return &watch{
		EnqueueRequestsFromMapFunc: handler.EnqueueRequestsFromMapFunc{
			ToRequests: handler.ToRequestsFunc(
				func(object handler.MapObject) []reconcile.Request {
					labels := object.Meta.GetLabels()
					if labels[common.TypeLabelName] == Type && labels[LicenseLabelType] == string(LicenseTypeEnterprise) {
						return fn.Map(object)
					}
					return nil
				}),
		},
	}
}
```

 But it does not seem to be used. We should check if it's a miss.

Also I'm wondering if we should not also trigger a reconciliation when `labels[LicenseLabelType] == string(LicenseTypeEnterpriseTrial)` ?

**Comment by barkbay** [comment]
TL;DR: This function is not required for now and can be removed. However we should revisit this issue when an other Elastic Stack application needs to reconcile on license changes. 

We discussed it out of band with @pebrc and for now this function is not required since license changes are already triggering Elasticsearch reconciliation loops with the help of [reconcileRequestsForAllClusters](https://github.com/elastic/cloud-on-k8s/blob/e52945b66f33debc67c0c1028714eec3216edfed/pkg/controller/license/list.go#L15).

We also don't think that there is a need in the short term as we don't plan to add a new controller _(or feature on an existing Elastic Stack controller)_ which would require to reconcile immediately on such an event.

A downside of the current code is that it is hard to manage errors in the handler (for example if the Kubernetes client returns an error), the only option in that situation is to silently drop the event: https://github.com/elastic/cloud-on-k8s/blob/e52945b66f33debc67c0c1028714eec3216edfed/pkg/controller/license/license_controller.go#L125-L129
